### ISSUE TITLE
Update to use Kotlin 1.6

### ISSUE THREAD

**Issue by carloshwa** [Stuie/papercut#17]
👋  After updating our app to Kotlin 1.6, we started getting the following build warning:
```
The following annotation processors are not incremental: papercut-compiler-0.9.1.jar (ie.stu:papercut-compiler:0.9.1).
Make sure all annotation processors are incremental to improve your build speed.
```

It seems incremental builds of the app are now performing a full build because of this. I think updating this library to use Kotlin 1.6 will resolve this.

**Comment by Stuie** [comment]
Thanks for the issue.

The annotation processor in Papercut needs to be updated to specify what type of [incremental processor](https://docs.gradle.org/current/userguide/java_plugin.html#sec:incremental_annotation_processing) it is (if any). This change is specific to Gradle, not the Kotlin version.

I've got a change in a local branch that I'll try to get pushed out early next week, once I've been able to test it properly.
